src/app/learnocaml_index_main.ml

```diff
@@ -85,8 +87,76 @@ let (exercise_filter_signal: string option React.signal), set_exercise_filter =
 let (exercise_sort_signal: exercise_ordering React.signal), set_exercise_sort =
   React.S.create By_category
 
+let (expand_state: string list React.signal), set_expand_state =
+  React.S.create []
+
+let encode str =
+  Re.Pcre.substitute ~rex:(Re.Pcre.regexp ",") ~subst:(fun _ -> "-c") (
+      Re.Pcre.substitute ~rex:(Re.Pcre.regexp "&") ~subst:(fun _ -> "-a") (
+          Re.Pcre.substitute ~rex:(Re.Pcre.regexp "=") ~subst:(fun _ -> "-e") (
+              Re.Pcre.substitute ~rex:(Re.Pcre.regexp "-") ~subst:(fun _ -> "--") str)))
+let decode str =
+  Re.Pcre.substitute ~rex:(Re.Pcre.regexp "--") ~subst:(fun _ -> "-") (
+      Re.Pcre.substitute ~rex:(Re.Pcre.regexp "-e") ~subst:(fun _ -> "=") (
+          Re.Pcre.substitute ~rex:(Re.Pcre.regexp "-a") ~subst:(fun _ -> "&") (
+              Re.Pcre.substitute ~rex:(Re.Pcre.regexp "-c") ~subst:(fun _ -> ",") str)))
+
+let rec update_expand ?value fragment =
```

Comment:
It appears in your PR, there is no additional `(* comments *)` while IMO it'd be a good idea to:

1. Add a small global comment, for example before `encode`, that briefly explains the feature added in the PR, its design, and the need to add the encode/decode mutual bijections…

2. Add individual comments of one line or so **just before each new function**, to explain their usefulness (except if their semantics is trivial, e.g., no need to document the `join` local function):
   * if the new function at stake is public (it has a `val …` in the `.mli`): document the function **in the .mli** only is OK
   * if the new function at stake is private (only in the `.ml`): document the function **in the .ml**